src/charm.py

```diff
@@ -1,103 +1,217 @@
+            "HTTP_PROXY": "http://squid.internal:3128",
+            "HTTPS_PROXY": "http://squid.internal:3128",
+            "NO_PROXY": "127.0.0.1,localhost,::1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.canonical.com,.launchpad.net,.internal,.jujucharms.com,temporal-k8s,minio",
+            "http_proxy": "http://squid.internal:3128",
+            "https_proxy": "http://squid.internal:3128",
+            "no_proxy": "10.0.0.0/8,localhost,127.0.0.1,.internal,.cluster.local,.local,.svc,airbyte-*,temporal-k8s,minio",
+            "JAVA_TOOL_OPTIONS": "-Dhttp.proxyHost=squid.internal -Dhttp.proxyPort=3128 -Dhttps.proxyHost=squid.internal -Dhttps.proxyPort=3128 -Dhttp.nonProxyHosts=10.0.0.0|localhost|127.0.0.1|*.internal|*.cluster.local|*.local|*.svc|airbyte-*|temporal-k8s|minio",
+            "JOB_DEFAULT_ENV_http_proxy": "http://squid.internal:3128",
+            "JOB_DEFAULT_ENV_https_proxy": "http://squid.internal:3128",
+            "JOB_DEFAULT_ENV_no_proxy": "127.0.0.1,localhost,::1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.canonical.com,.launchpad.net,.internal,.jujucharms.com,temporal-k8s,minio",
+            "JOB_DEFAULT_ENV_HTTP_PROXY": "http://squid.internal:3128",
+            "JOB_DEFAULT_ENV_HTTPS_PROXY": "http://squid.internal:3128",
+            "JOB_DEFAULT_ENV_NO_PROXY": "127.0.0.1,localhost,::1,10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,.canonical.com,.launchpad.net,.internal,.jujucharms.com,temporal-k8s,minio",
+            "JOB_DEFAULT_ENV_JAVA_TOOL_OPTIONS": "-Dhttp.proxyHost=squid.internal -Dhttp.proxyPort=3128 -Dhttps.proxyHost=squid.internal -Dhttps.proxyPort=3128 -Dhttp.nonProxyHosts=10.0.0.0|localhost|127.0.0.1|*.internal|*.cluster.local|*.local|*.svc|airbyte-*|temporal-k8s|minio",
         }
 
+    def _update(self, event):
+        """Update the Temporal server configuration and replan its execution.
+
+        Args:
+            event: The event triggered when the relation changed.
+        """
+        # TODO (kelkawi-a): populate database connection from PostgreSQL relation
+        if not self._state.database_connections:
+            self.unit.status = BlockedStatus("missing db relation")
+            return
+
+        env = self._create_env()
+        self.model.unit.set_ports(CONNECTOR_BUILDER_SERVER_API_PORT)
+        self.model.unit.set_ports(INTERNAL_API_PORT)
+
+        for service in CONTAINERS:
+            container = self.unit.get_container(service)
+            if not container.can_connect():
+                event.defer()
+                return
+
+            if service == "airbyte-pod-sweeper":
+                script_path = Path(__file__).parent / "scripts/pod-sweeper.sh"
+
+                with open(script_path, "r") as file_source:
+                    logger.info("pushing pod-sweeper script...")
+                    container.push(
+                        "/pod-sweeper.sh", file_source, make_dirs=True, permissions=0o755
```

Comment:
could this be pushed in `airbyte-app/bin/` so you wouldn't have to do treat it differently above?

Reply:
Good idea